starknetdata/starknetdata.go

```diff
@@ -11,7 +11,8 @@ import (
 //
 //go:generate mockgen -destination=../mocks/mock_starknetdata.go -package=mocks github.com/NethermindEth/juno/starknetdata StarknetData
 type StarknetData interface {
-	BlockByNumber(ctx context.Context, blockNumber uint64) (*core.Block, error)
+	BlockByNumber(ctx context.Context, blockID uint64) (*core.Block, error)
+	BlockByID(ctx context.Context, blockID string) (*core.Block, error)
```

Comment:
Instead of a `string`, we should use a more well defined type here to avoid any confusion on what the valid inputs are. Let's either make this `BlockByTag` and define a `BlockTag` enum with Pending and Latest values

or let's just jave `BlockLatest` that does not need any arguments

Reply:
Done